src/utils/analysis-utils.ts

```diff
@@ -648,3 +676,59 @@ export class BaselineLayerResult {
       : `${this.getHazardLayer().title} (${this.statistic})`;
   }
 }
+
+export class PolygonAnalysisResult {
+  key: number = Date.now();
+  featureCollection: FeatureCollection;
+  tableData: TableRow[];
+  tableColumns: string[];
+
+  statistic: 'area' | 'percentage';
+  threshold?: ThresholdDefinition;
+  legend: LegendDefinition;
+  legendText: string;
+  hazardLayerId: WMSLayerProps['id'];
+  adminLevel: AdminLevelType;
+
+  constructor(
+    tableData: TableRow[],
+    tableColumns: string[],
+    featureCollection: FeatureCollection,
+    hazardLayer: WMSLayerProps,
+    adminLevel: AdminLevelType,
+    statistic: 'area' | 'percentage',
+    threshold?: ThresholdDefinition,
+  ) {
+    this.featureCollection = featureCollection;
+    this.tableData = tableData;
+    this.tableColumns = tableColumns;
+    this.statistic = statistic;
+    this.threshold = threshold;
+    this.adminLevel = adminLevel;
+
+    // color breaks from https://colorbrewer2.org/#type=sequential&scheme=Reds&n=5
+    this.legend = [
```

Comment:
Why not use % instead? -> 20, 40, ...

Reply:
(this would solve the legend issue)

Reply:
confirming it works
<img width="230" alt="Screen Shot 2022-03-31 at 10 43 18 PM" src="https://user-images.githubusercontent.com/4313463/161184934-5a04ae04-5ac1-4c00-9dd8-9a6a47e555fa.png">